droopler.install

```diff
@@ -1818,3 +1822,86 @@ function droopler_update_8127(&$sandbox) {
 
 }
 
+/**
+ * Droopler 2.2 Migrate layout settings.
+ */
+function droopler_update_8128(&$sandbox) {
+
+  $bundles = [
+    'd_p_tiles',
+    'd_p_side_tiles',
+    'd_p_side_image',
+    'd_p_side_embed',
+    'd_p_side_form',
+  ];
+  $paragraphs_per_batch = 25;
+  if (!isset($sandbox['current'])) {
+    $sandbox['current'] = 0;
+  }
+
+  $paragraphs_ids = \Drupal::entityQuery('paragraph')
+    ->condition('type', $bundles, 'IN')
+    ->range($sandbox['current'], $paragraphs_per_batch)
+    ->execute();
+
+  foreach ($paragraphs_ids as $id) {
+    $paragraph = \Drupal::entityTypeManager()
+      ->getStorage('paragraph')
+      ->load($id);
+
+    $settings = $paragraph->hasField('field_d_settings') && $paragraph->get('field_d_settings') ? $paragraph->get('field_d_settings')->value : "{}";
+    $json = json_decode($settings);
```

Comment:
Pytanie czy dałoby się zastąpić manipulowanie w json przez używanie helpera od d_settings.

Reply:
oo w sumie racja, to chyba nie jest konieczne :thinking:

Reply:
będzie do poprawienia przy okazji alterowania, szkoda już tu na to czasu